domain/src/test/java/org/oppia/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -95,8 +122,392 @@ class ProfileManagementControllerTest {
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testUpdateLastLoggedIn_addProfiles_updateWithBadProfileId_checkIsFailure() = runBlockingTest(coroutineContext) {
+    addTestProfiles()
+    advanceUntilIdle()
+
+    val profileId = ProfileId.newBuilder().setInternalId(6).build()
+    profileManagementController.updateLastLoggedIn(profileId).observeForever(mockUpdateResultObserver)
+    advanceUntilIdle()
+
+    verify(mockUpdateResultObserver, atLeastOnce()).onChanged(updateResultCaptor.capture())
+    assertThat(updateResultCaptor.value.isFailure()).isTrue()
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testDeleteProfile_addProfiles_deleteProfile_checkIsSuccessful() = runBlockingTest(coroutineContext) {
+    addTestProfiles()
+    advanceUntilIdle()
+
+    val profileId = ProfileId.newBuilder().setInternalId(2).build()
+    profileManagementController.deleteProfile(profileId).observeForever(mockUpdateResultObserver)
+    advanceUntilIdle()
+    profileManagementController.getProfile(profileId).observeForever(mockProfileObserver)
+
+    verify(mockUpdateResultObserver, atLeastOnce()).onChanged(updateResultCaptor.capture())
+    verify(mockProfileObserver, atLeastOnce()).onChanged(profileResultCaptor.capture())
+    assertThat(updateResultCaptor.value.isSuccess()).isTrue()
+    assertThat(profileResultCaptor.value.isSuccess()).isTrue()
+    assertThat(profileResultCaptor.value.getOrThrow().name).isEmpty()
+    assertThat(File(getAbsoluteDirPath("2")).isDirectory).isFalse()
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testDeleteProfile_addProfiles_deleteProfiles_addProfile_checkIdIsNotReused() = runBlockingTest(coroutineContext) {
+    addTestProfiles()
+    advanceUntilIdle()
+
+    val profileId3 = ProfileId.newBuilder().setInternalId(3).build()
+    val profileId4 = ProfileId.newBuilder().setInternalId(4).build()
+    profileManagementController.deleteProfile(profileId3)
+    profileManagementController.deleteProfile(profileId4)
+    profileManagementController.addProfile("John", "321", null, true)
+    advanceUntilIdle()
+    profileManagementController.getProfiles().observeForever(mockProfilesObserver)
+
+    verify(mockProfilesObserver, atLeastOnce()).onChanged(profilesResultCaptor.capture())
+    assertThat(profilesResultCaptor.value.isSuccess()).isTrue()
+    val profiles = profilesResultCaptor.value.getOrThrow().sortedBy { it.id.internalId }
+    assertThat(profiles.size).isEqualTo(4)
+    assertThat(profiles[profiles.size - 2].name).isEqualTo("Ben")
+    assertThat(profiles.last().name).isEqualTo("John")
+    assertThat(profiles.last().id.internalId).isEqualTo(5)
+    assertThat(File(getAbsoluteDirPath("3")).isDirectory).isFalse()
+    assertThat(File(getAbsoluteDirPath("4")).isDirectory).isFalse()
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testDeleteProfile_addProfiles_deleteProfiles_restartApplication_checkIsSuccessful() = runBlockingTest(coroutineContext) {
```

Comment:
What is successful? I can't tell from the test name whether it's supposed to be add or delete profiles. Can you clarify here?

Reply:
Updated test names to include the fail or success operation.